openshift_cli_installer/utils/gcp_utils.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+import shutil
+from pathlib import Path
+import tempfile
+from pyaml_env import parse_config
+
+from simple_logger.logger import get_logger
+from openshift_cli_installer.utils.const import GCP_STR
+
+LOGGER = get_logger(name=__name__)
+TMP_FILE_NAME = tempfile.NamedTemporaryFile(suffix="installer.json").name
+
+
+def get_gcp_params():
+    gcp_params = {}
+    gcp_params["gcp_sa_file_dir"] = os.path.join(os.path.expanduser("~"), ".gcp")
+    gcp_params["openshift_installer_gcp_sa_file_path"] = os.path.join(
+        gcp_params["gcp_sa_file_dir"], "osServiceAccount.json"
+    )
+    gcp_params["backup_existing_gcp_sa_file_path"] = TMP_FILE_NAME
+    return gcp_params
+
+
+def set_gcp_configuration(**kwargs):
+    """
+    Saves provided GCP Service Account file at location '~/.gcp/osServiceAccount.json'
+
+    If file already exists at this path, it will be copied to tmp file first.
+
+    """
+    user_input = get_user_input(**kwargs)
+    if is_platform_gcp(user_input["clusters"]):
+        gcp_params = get_gcp_params()
+        if os.path.exists(gcp_params["openshift_installer_gcp_sa_file_path"]):
```

Comment:
This function can return `gcp_params` and you can add another key to it if backup was created, then `gcp_params` should be passed to `restore_gcp_configuration`

Reply:
updated the code, please review again